homeassistant/components/tuya/number.py

```diff
@@ -373,8 +373,14 @@ def __init__(
             if self.device_class:
                 self._attr_icon = None
 
+            # Found unit of measurement, use the standardized Unit
+            # Use the target conversion unit (if set)
+            self._attr_native_unit_of_measurement = (
+                self._uom.conversion_unit or self._uom.unit
+            )
```

Comment:
This was missing in #73483, I'm not sure if this is correct though.

Reply:
Yes! Thanks 👍